src/App.js

```diff
@@ -1,7 +1,37 @@
-import "./App.css";
+import React from "react";
+import ItemList from "./Component/ItemList";
+import CheckOut from "./Component/CheckOut";
+import Cart from "./Component/Cart"
 
-const App = () => {
-  return <h1>Hello, world!</h1>;
-};
 
-export default App;
+
+class App extends React.Component {
+   state ={cart:[], subtotal:0}
+
+
+ addCart = (item) => {
+   this.setState((prevState) => {
+    this.subtotal(item.price)
+     return {cart:[item, ...prevState.cart]}
+   })
```

Comment:
When using this form of setState, the callback you pass should not have any side effects. (calling this.subtotal inside the callback is an example of a side effect). This is causing the price of the item to be added to the subtotal twice each time a user hits an Add to Cart button.

Suggestion: When you need to update two different state values, you can do it with a single function like this:

`addCart = (item) => {
    this.setState((prevState) => {
        return {
            subtotal: prevState.subtotal + item.price,
            cart: [...prevState.cart, item]
        }
    })
}`